### PR TITLE
Connector docs

### PR DESCRIPTION
**Pull request by Licenser**
2nd part of moving connector docs to the main repo. With this, all reference documentation lives along with the code

### CODE REVIEW

**Comment by mfelsche** [comment]
I like, but this needs some changes in the markdown to show a proper title in the left navbar. Both for the index page and the single connector pages.
This is how it looks right now in the preview:

image

File: docs/reference/connectors/bench.md [hunk 1]
```diff
@@ -1,194 +0,0 @@
----
-sidebar_label: bench (Benchmarking)
```
Comment (mfelsche) [inline docs/reference/connectors/bench.md:2]: this is what is missing.

File: docs/reference/connectors/_category_.yaml [hunk 2]
```diff
@@ -1 +0,0 @@
-position: 1
```
Comment (mfelsche) [inline docs/reference/connectors/_category_.yaml:1]: I think this file is what makes docusaurus display the stuff nicely in the left bar. Currently it adds an `index` entry and doesn't take `index.md` as what should be displayed at the top-level.

**Comment by mfelsche** [comment]
Also admonitions in the generated docs are not resolved anymore :(

File: docs/reference/connectors/bench.md (hunk 1 above)
Comment (Licenser) [reply docs/reference/connectors/bench.md:2]: the problem with introducing those headers is that we're now having docusaurus specific syntax in the comments that'd probably not play nice with rust doc generation, is having the direct names really a big issue?

File: docs/reference/connectors/_category_.yaml (hunk 2 above)
Comment (Licenser) [reply docs/reference/connectors/_category_.yaml:1]: Nope, this file just moved to now be part of the docs, the reason is the headers.

**Comment by Licenser** [comment]
most of this should be fixed in https://github.com/tremor-rs/tremor-runtime/pull/2332

**Comment by Licenser** [comment]
@mfelsche most things addressed :) with the exception of the naming. I think that should be discussed separately as it would need to be uniform between all resources (codecs, connectors, etc ...)

**Review by Agares — APPROVED** [review]

**Review by darach — DISMISSED** [review]
LGTM 👍

**Review by Agares — APPROVED** [review]

**Pull request merged**